### PR TITLE
BP-2297 refactor response handlers

### PR DESCRIPTION
**Pull request by LucianTuriacArnia**

### CODE REVIEW

File: Controller/Applepay/SaveOrder.php [hunk 1]
```diff
@@ -203,21 +199,27 @@ private function submitQuote($quote, $extra)
...
-                ->setCustomerIsGuest(true)
-                ->setCustomerGroupId(\Magento\Customer\Model\Group::NOT_LOGGED_IN_ID);
-        }
+        try {
+            if (!($this->customerSession->getCustomer() && $this->customerSession->getCustomer()->getId())) {
+                $quote->setCheckoutMethod('guest')
+                    ->setCustomerId(null)
+                    ->setCustomerEmail($quote->getShippingAddress()->getEmail())
+                    ->setCustomerIsGuest(true)
+                    ->setCustomerGroupId(\Magento\Customer\Model\Group::NOT_LOGGED_IN_ID);
```
Comment (harli91) [inline Controller/Applepay/SaveOrder.php:208]: Maybe use namespaces here

File: Gateway/Request/BasicParameter/SaveOrderBeforeDataBuilder.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        $store = $order->getStoreId();
+
+        if ($this->configProviderAccount->getCreateOrderBeforeTransaction($store)) {
+            $newStatus = $this->configProviderAccount->getOrderStatusNew($store);
+            $orderState = 'new';
+            if (!$newStatus) {
+                $newStatus = $order->getConfig()->getStateDefaultStatus($orderState);
+            }
+
+            $order->setState($orderState);
```
Comment (harli91) [inline Gateway/Request/BasicParameter/SaveOrderBeforeDataBuilder.php:45]: why do we need to set the status/state here?

File: Gateway/Response/AuthorizeTransactionIdHandler.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace Buckaroo\Magento2\Gateway\Response;
+
+use Buckaroo\Transaction\Response\TransactionResponse;
+use Magento\Sales\Api\Data\OrderPaymentInterface;
+use Buckaroo\Magento2\Gateway\Helper\SubjectReader;
+use Magento\Payment\Gateway\Response\HandlerInterface;
+use Magento\Sales\Model\Order\Payment;
+
+class AuthorizeTransactionIdHandler extends TransactionIdHandler
```
Comment (harli91) [inline Gateway/Response/AuthorizeTransactionIdHandler.php:11]: what's the use of this class?

File: Gateway/Response/ConsumerMessageHandler.php [hunk 4]
```diff
@@ -3,11 +3,25 @@
 namespace Buckaroo\Magento2\Gateway\Response;
 
 use Buckaroo\Magento2\Gateway\Helper\SubjectReader;
-use Buckaroo\Transaction\Response\TransactionResponse;
+use Magento\Framework\Message\ManagerInterface as MessageManager;
 use Magento\Payment\Gateway\Response\HandlerInterface;
 
-class ConsumerMessageHandler extends AbstractResponseHandler implements HandlerInterface
+class ConsumerMessageHandler implements HandlerInterface
```
Comment (harli91) [inline Gateway/Response/ConsumerMessageHandler.php:9]: where is this class used and why?

File: Gateway/Request/BasicParameter/SaveOrderBeforeDataBuilder.php (hunk 2 above)
Comment (LucianTuriacArnia) [reply Gateway/Request/BasicParameter/SaveOrderBeforeDataBuilder.php:45]: The logic was copied from here: app/code/Buckaroo/Magento2/Gateway/Http/TransactionBuilder/Order.php:66
And orderStatus is set from configuration. For orderState I don't have explanation why is set on "New" if any order start with the state New.

File: Gateway/Response/AuthorizeTransactionIdHandler.php (hunk 3 above)
Comment (LucianTuriacArnia) [reply Gateway/Response/AuthorizeTransactionIdHandler.php:11]: When we make an order with KlarnaKP we shouldn't close transaction. app/code/Buckaroo/Magento2/Model/Method/AbstractMethod.php:938

image

File: Gateway/Response/ConsumerMessageHandler.php (hunk 4 above)
Comment (LucianTuriacArnia) [reply Gateway/Response/ConsumerMessageHandler.php:9]: It is used in <virtualType name="SepaDirectDebitHandler"> and set the success message on success page with what we receive from plaza

Comment (harli91) [reply Gateway/Response/ConsumerMessageHandler.php:9]: why not use the MessageManager directly ?

**Review by harli91 — APPROVED** [review]

**Pull request merged**